cranelift-codegen/src/redundant_reload_remover.rs

```diff
@@ -0,0 +1,462 @@
+                    let src_reg = self.reg_of_value(*src_value);
+                    if let Some(offset) = slot.offset {
+                        let size = slot.size;
+                        self.state.env.invalidate_by_offset(offset, size);
+                        self.state.env.bind(src_reg, *src_value, offset, size);
+                    } else {
+                        panic!("cleanup_ebb: spill with no offset");
+                    }
+                }
+                InstructionData::Unary {
+                    opcode: Opcode::Fill,
+                    arg: src_value,
+                } => {
+                    // Extract: (dst_value, dst_regunit, offset, size)
+                    // Invalidate: (offset, size)
+                    // Add new: {dst_regunit -> (dst_value, offset, size)}
+                    let slot = self.slot_of_value(*src_value);
+                    let dst_value = self.cur.func.dfg.inst_results(inst)[0];
+                    let dst_reg = self.reg_of_value(dst_value);
+                    if let Some(offset) = slot.offset {
+                        let size = slot.size;
+                        if self.state.env.has_exact_binding(dst_reg, offset, size) {
+                            // This instruction is an exact copy of a fill we saw earlier, and the
+                            // loaded value is still valid.  So we'll schedule this instruction
+                            // for deletion (below).  We also update the binding so as to
+                            // associate this instruction's result SSA name with it.
+                            replacement = Some((true, *src_value));
+                            self.state.env.bind(dst_reg, dst_value, offset, size);
+                        } else if let Some((reg2, val2)) =
+                            self.state
+                                .env
+                                .has_inexact_binding(&self.reginfo, dst_reg, offset, size)
+                        {
+                            // This fill is from the required slot, but into a different register
+                            // `reg2`.  So replace it with a copy from `reg2` to `dst_reg` and
+                            // update `dst_reg`s binding to make it the same as `reg2`'s, so as to
+                            // maximise the chances of future matches after this instruction.
+                            debug_assert!(reg2 != dst_reg);
+                            replacement = Some((false, val2));
+                            self.state.env.bind(dst_reg, dst_value, offset, size);
+                        } else {
+                            // This fill creates some new binding we don't know about.  Update
+                            // `env` to track it.
+                            self.state.env.invalidate_by_offset(offset, size);
+                            self.state.env.bind(dst_reg, dst_value, offset, size);
```

Comment:
nit: this `bind` line is redundant in the three branches of this if/else. Could we move it outside the if/else, please?